src/managers/SessionLockManager.cpp

```diff
@@ -99,18 +99,20 @@ void CSessionLockManager::onNewSessionLock(SP<CSessionLock> pLock) {
                 if (!g_pSessionLockManager || g_pSessionLockManager->clientLocked() || g_pSessionLockManager->clientDenied())
                     return;
 
-                if (g_pCompositor->m_unsafeState || !g_pCompositor->m_aqBackend->hasSession() || !g_pCompositor->m_aqBackend->session->active) {
-                    // Because the session is inactive, there is a good reason for why the client did't receive locked or denied.
+                if (!g_pSessionLockManager->m_sessionLock || !g_pSessionLockManager->m_sessionLock->lock)
+                    return;
+
+                if (!g_pCompositor->m_dpmsStateOn || g_pCompositor->m_unsafeState || !g_pCompositor->m_aqBackend->hasSession() || !g_pCompositor->m_aqBackend->session->active) {
```

Comment:
m_dpmsStateOn should be nuked, it's dumb as dpms is per monitor, so a global doesnt make sense.